File: gang/week12/기사단원의 무기.java
class Solution {
    public int solution(int number, int limit, int power) {
        int answer = 0;
        for(int i=1; i<=number; i++) {
            if(divisor(i) > limit) {    // 공격력 제한보다 높으면
                answer += power;    // 제한 공격력 추가
            }
            else {
                answer += divisor(i);
            }
        }
        return answer;
    }

    public int divisor(int num) {
        int cnt = 0;
        for (int i = 1; i * i <= num; i++) {    // 제곱근 사용
            if(i*i == num) {
                cnt++;
            }
            else if (num % i == 0) {
                cnt += 2;
            }
        }
        return cnt;
    }
}

File: jeonghyeon/12week/기사단원의 무기.js
function solution(number, limit, power) {
    var answer = 0;
    
    for(let i = 1; i <= number; i++) {
        let div = 0;
        for(let j = 1; j <= Math.sqrt(i); j++) {
            if(i % j == 0) {
                div += 2;
                if(i / j == j) {
                    div--;
                }
            }
        }
        if(div > limit) {
            answer += power;
        } else {
            answer += div;
        }
    }
    return answer;
}

// number 돌면서 약수 개수 구함
// 약수 개수 > limit answer += power
// 약수 개수 <= limit answer += 약수 개수
// return answer